icu4c/source/common/uloc.cpp

```diff
@@ -1478,7 +1478,7 @@ _canonicalize(const char* localeID,
               uint32_t options,
               UErrorCode* err) {
     int32_t j, fieldCount=0, scriptSize=0, variantSize=0;
-    PreflightingLocaleIDBuffer tempBuffer;
+    PreflightingLocaleIDBuffer tempBuffer;  // if localeID has a BCP47 extension, tmpLocaleID points to this
```

Comment:
Before the previous line (that is, as the first thing in the function body) please insert
```
if (U_FAILURE(*err)) { return 0; }
```